// Kaesi Manakkal
syntax = "proto3";

package project1;

// The search engine service definition.
service ImgSearchEngine {
    // make a search
    rpc MakeQuery (Query) returns (Image) {}
}

// The request message containing the query to search.
message Query {
    string query = 1;       // keyword query (the class of image we want)
}

// The response image containing the result
message Image {
    bytes image = 1;        // this contains the image as bytes
    int32 error = 2;        // a possible error code (404 = not found; 200=ok; 500=other fatal error (ex. db doesn't exist))
    int32 len = 3;          // the length of the image bytes (image file size in bytes)
}
